main.go

```diff
@@ -77,11 +77,28 @@ func mainWithExitCode() int {
 	logrus.Debug("GOLLUM STARTING")
 	defer logrus.Debug("GOLLUM STOPPED")
 
-	config := readConfig()
```

Comment:
Why move all this testing and flag check code back to the main function?
The main function should not check for anything but boolean flags. Everything else should go into sub functions.